### PR TITLE
CLN: tighten noqas

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback do build is green here

**Comment by jreback** [comment]
wow haven't seen green in a while
thxs!

**Comment by phofl** [comment]
I think we should backport the doc fix?

**Comment by jbrockmendel** [comment]
sure.  its just the one line

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -1849,5 +1849,5 @@ def union_with_duplicates(lvals: ArrayLike, rvals: ArrayLike) -> ArrayLike:
     unique_array = ensure_wrapped_if_datetimelike(unique_array)
 
     for i, value in enumerate(unique_array):
-        indexer += [i] * int(max(l_count[value], r_count[value]))
+        indexer += [i] * int(max(l_count.at[value], r_count.at[value]))
```
Comment (jorisvandenbossche) [inline pandas/core/algorithms.py:1852]: Same question as in the other PR: what's the reason for this change? It shouldn't influence behaviour, but just being more explicit about label-based?

**Comment by jbrockmendel** [comment]
More explicit, yes
